src/NLog/LayoutRenderers/Wrappers/ReplaceNewLinesLayoutRendererWrapper.cs

```diff
@@ -82,19 +68,62 @@ protected override void RenderInnerAndTransform(LogEventInfo logEvent, StringBui
             Inner?.Render(logEvent, builder);
             if (builder.Length > orgLength)
             {
-                var containsNewLines = builder.IndexOf('\n', orgLength) >= 0;
-                if (containsNewLines)
+                int lineEndIndex = IndexOfLineEndCharacters(builder, orgLength);
+                if (lineEndIndex > -1)
                 {
-                    string str = builder.ToString(orgLength, builder.Length - orgLength)
-                                        .Replace(WindowsNewLine, _replaceWithNewLines ? UnixNewLine : Replacement)
-                                        .Replace(UnixNewLine, Replacement);
-
+                    orgLength = lineEndIndex > orgLength ? lineEndIndex - 1 : orgLength;
+                    string str = builder.ToString(orgLength, builder.Length - orgLength);
                     builder.Length = orgLength;
-                    builder.Append(str);
+
+                    bool ignoreNewLine = false;
+                    foreach (var chr in str)
+                    {
+                        if (IsLineEndCharacter(chr))  // switches on chr, and allow method-reuse
+                        {
+                            if (!ignoreNewLine || chr != '\n')
+                            {
+                                builder.Append(Replacement);
+                            }
+                            ignoreNewLine = chr == '\r';
+                        }
+                        else
+                        {
+                            builder.Append(chr);
+                            ignoreNewLine = false;
+                        }
+                    }
+                }
+            }
+        }
+
+        private static bool IsLineEndCharacter(char ch)
```

Comment:
```csharp
        private static bool IsLineEndCharacter(char ch)
        {
            switch (ch)
            {
                case '\r':
                case '\n':
                case '\u000C':  // FORM FEED (FF)
                case '\u0085':  // NEXT LINE (NEL)
                case '\u2028':  // LINE SEPARATOR
                case '\u2029': // PARAGRAPH SEPARATOR
                    return true;
            }
            return false;
        }
```
Believe this will be my last nitpick. Think the updated code is a good improvement, and thank you again for the benchmark.